sensing/src/main/java/com/google/android/sensing/upload/SensorDataUploadWorker.kt

```diff
@@ -59,39 +55,41 @@ class SensorDataUploadWorker(appContext: Context, workerParams: WorkerParameters
   }
 
   override suspend fun doWork(): Result {
+    if (runAttemptCount == inputData.getInt(MAX_RETRIES_ALLOWED, 0)) return Result.failure()
```

Comment:
nit: Should we make it something like `runAttemptCount >= inputData.getInt(MAX_RETRIES_ALLOWED, 0)` to be on a safer side?